### PR TITLE
Issue #30 #33 Solved 

### PR DESCRIPTION
**Pull request by anushkasaxena07**
## PR Description

The issue with the existing code is that it directly inserts user input into SQL queries, making it vulnerable to SQL injection attacks. To address this, the code needs to be refactored to use prepared statements with parameter binding. Prepared statements separate SQL logic from data, preventing malicious SQL code injection.

Solution:

Here's how to solve the issue:

1. save_progress() function:
   - Refactor the code to use a prepared statement with parameter binding.
   - Bind parameters to the statement to ensure safe execution.
   - Execute the statement.

2. delete_progress() function:
   - Refactor to use a prepared statement with parameter binding.
   - Bind the id parameter to the statement.
   - Execute the statement.

3. save_restriction() function:
   - Loop through the $rid array to handle multiple rows.
   - Use prepared statements with parameter binding for both INSERT and UPDATE operations.
   - Execute the statements within the loop.

4. save_evaluation() function:
   - Use a prepared statement with parameter binding to insert data into evaluation_list.
   - Loop through $qid to handle multiple question evaluations.
   - Use prepared statements with parameter binding to insert data into evaluation_answers.
   - Execute the statements within the loop.

5. get_class() function:
   - Refactor to use a prepared statement with parameter binding.
   - Bind parameters for $fid and academic_id.
   - Execute the statement and fetch results.

6. get_report() function:
   - Refactor to use a prepared statement with parameter binding.
   - Bind parameters for academic_id, faculty_id, subject_id, and class_id.
   - Execute the statement to get evaluation answers.
   - Use a prepared statement to get the total answered rows.
   - Calculate percentages and return the JSON-encoded data.

By implementing these changes, it ensures that user input is properly sanitized, and the application is protected from SQL injection attacks. Prepared statements with parameter binding handle user data safely, preventing malicious SQL injection attempts.

Related Issues: Issue for which you are raising a PR for Avoiding SQL injections using prepared statements of MySQLi #30
Closes https://github.com/SaiyamTuteja/Faculty_Evaluation_System/issues/30
## Issue
#30 
## Issue
#33 
 ## PR Description

1. I modified the PHP script to include a condition for handling POST requests sent by the form.
2. Upon receiving the form data, I perform checks to ensure the email doesn't already exist for another user using a SELECT query.
3. If the email does exist, I echo 2 to indicate this condition.
4. If the update is successful, I echo 1.
5. If there's an error during the update process, I echo 0.
This implementation ensures that the backend properly handles the scenario where the username already exists when updating user details.

## Checklist
[ x] I have gone through the [contributing guide](https://github.com/SaiyamTuteja/Faculty_Evaluation_System/blob/main/CONTRIBUTING.md)
[ x] I have updated my branch and synced it with project main branch before making this PR
[x ] Is this a bug fix/enhancement/documentation changes
[ x] Part of GSSOC
[ x] Tested for any breaking changes
[x ] Other relevant checks completed
## Undertaking
I declare that:

The content I am submitting is original and has not been plagiarized.
No portion of the work has been copied from any other source without proper attribution.
The work has been checked for plagiarism, and I assure its authenticity.
I understand that any violation of this undertaking may have legal consequences that I will bear and could result in the withdrawal of any recognition associated with the work.

[ x] I Agree

### CODE REVIEW

File: admin_class.php [hunk 1]
```diff
@@ -1,690 +1,1153 @@
+function login(){
```
Comment (aswinikalyan30) [inline admin_class.php:1]: I don't see the initial lines fo the session_start() added to this file - Please cross check and add it:
```
<?php
session_start();
ini_set('display_errors', 1);
Class Action {
	private $db;

	public function __construct() {
		ob_start();
   	include 'db_connect.php';
    
    $this->db = $conn;
	}
	function __destruct() {
	    $this->db->close();
	    ob_end_flush();
	}
```

**Review by aswinikalyan30 — CHANGES_REQUESTED** [review]

**Comment by aswinikalyan30** [comment]
Needs more work - Please let me know if you have any trouble setting it up on your local and we can chat over discord

**Comment by anushkasaxena07** [comment]
@aswinikalyan30 sorry for my negligence please have a look now 
is it fine?

File: admin_class.php [hunk 2]
```diff
@@ -1,690 +1,1172 @@
...
+
+    // Check if system_settings table has data
+    $chk = $this->db->query("SELECT * FROM system_settings");
+    if($chk->num_rows > 0){
+        $data_types = str_repeat("s", count($_POST)); // Assuming all parameters are strings
+        $data_values = array_values($_POST);
+        
+        // Prepare the update query
+        $stmt = $this->db->prepare("UPDATE system_settings SET $data WHERE id = ?");
+        $stmt->bind_param($data_types."i", ...$data_values, $chk->fetch_array()['id']);
```
Comment (aswinikalyan30) [inline admin_class.php:353]: Fatal error: Cannot use positional argument after argument unpacking in /Applications/MAMP/htdocs/Faculty_Evaluation_System-main/admin_class.php on line 354
Getting this error

Comment (aswinikalyan30) [reply admin_class.php:353]: Ran it on my local and got this error - How are you testing your changes?

File: admin_class.php [hunk 3]
```diff
@@ -1,690 +1,1172 @@
...
+
+    if($stmt->affected_rows > 0){
+        $stmt->close();
+        return 1;
+    }else{
+        $stmt->close();
+        return 0;
+    }
+}
+<?php
```
Comment (aswinikalyan30) [inline admin_class.php:1000]: Causing syntax issues on this line - remove it

File: manage_user.php [hunk 4]
```diff
@@ -1,92 +1,124 @@
...
-			}
-		})
-	})
+            reader.readAsDataURL(input.files[0]);
+        }
+    }
 
-</script>
\ No newline at end of file
+    $('#manage-user').submit(function(e){
```
Comment (aswinikalyan30) [inline manage_user.php:97]: This function calls an empty url - but I see the query is written on top - how are you testing any of your changes?
Going forward I need screenshots added to the PR to see what output you are seeing on screen

**Review by aswinikalyan30 — CHANGES_REQUESTED** [review]

**Comment by aswinikalyan30** [comment]
Hey @anushkasaxena07 - any progress?

**Comment by anushkasaxena07** [comment]
Still working on it

**Comment by aswinikalyan30** [comment]
Hey @anushkasaxena07 this has been open for too long, if you're not able to solve it will assign to someone else

**Comment by aswinikalyan30** [comment]
Closing this PR as no working changes are made for more than 2 weeks and is blocking contributions from others

**Pull request closed without merging**